### ISSUE TITLE
New QuickShell terminal unusable

### ISSUE THREAD

**Issue by i0ntempest** [nileshkhaire/QuickShell#4]
This is probably a configuration problem but I can't figure out where it is. When I start a new jshell instance, it looks like this and anything I type is not shown:
![Screen Shot 2022-07-01 at 22 24 41](https://user-images.githubusercontent.com/16017904/176983490-39692922-449b-4a68-953d-963c7331efc2.png)
If I hit enter a bunch of times:
![Screen Shot 2022-07-01 at 22 24 54](https://user-images.githubusercontent.com/16017904/176983496-35c7526a-13e6-4c1e-9365-68480cf94b7b.png)
But it does run the code I type in, I just can't see anything:
![Screen Shot 2022-07-01 at 22 25 24](https://user-images.githubusercontent.com/16017904/176983516-3ebb95f4-ecc5-4f83-9e63-817b0c953436.png)
If I start Eclipse with this terminal open, it becomes usable:
![image](https://user-images.githubusercontent.com/16017904/176983559-c2eb76de-6785-4085-820c-ffee64569d5f.png)

Any idea?

**Comment by nileshkhaire** [comment]
Hello @i0ntempest  May I know which Eclipse version and OS are you using ? Can you please try to upgrade or downgrade Eclipse version and see if it is working ?
I also don't understand, when you start Eclipse why 'QuickShell terminal' is replaced with 'i0ntempest.dynu.net' ? (See your last screenshot)

**Comment by i0ntempest** [comment]
> Hello @i0ntempest May I know which Eclipse version and OS are you using ? Can you please try to upgrade or downgrade Eclipse version and see if it is working ?
> I also don't understand, when you start Eclipse why 'QuickShell terminal' is replaced with 'i0ntempest.dynu.net' ? (See your last screenshot)

I’m using the latest eclipse. Not sure how to downgrade but I’ll try using a new installation/workspace. The domain name in the last screenshot is the machine’s host name.

**Comment by nileshkhaire** [comment]
If you click on the Help menu and then click on 'About Eclipse IDE' command , you can see  version as well as type of IDE (For example, Eclipse IDE for Java Developers, Eclipse IDE for Enterprise Java and Web developers,etc) 

Downgrade means if you can download old Eclipse version manually and try it. For example on this link you can download old version of eclipse. [Download Eclipse 2021-12] ( https://www.eclipse.org/downloads/packages/release/2021-12/r )

I have tested with Eclipse IDE for Enterprise Java and Web Developers (Version 2012-12) on Windows and it is working .

**Comment by i0ntempest** [comment]
Alright, I just tried downloading 2021-12 and creating a new workspace with it, then installed QuickShell. Still the same problem. I'm on macOS, jshell executable is selected as `/usr/bin/jshell`. Might be a platform specific problem?

**Comment by nileshkhaire** [comment]
This could be platform specific problem. After doing google search I found out that, other users also have this problem. See here : [Stackoverflow link ](https://stackoverflow.com/questions/64243467) 
Could you please also try to update the TM Terminal as specified in the above link? 

This could also mean that this is not a problem of QuickShell but a problem of TM Terminal which is used by QuickShell internally. 

Another thing is, QuickShell is capable of opening any executable not only Jshell. So if you go to QuickShell settings in Eclipse( Window menu-> Preferences -> QuickShell Settings -> Select jshell Executable) and pass another executable like python and try to start QuickShell in Eclipse Terminal. If you face the same issue then the issue is not related to QuickShell but related to TM terminal.

**Comment by nileshkhaire** [comment]
I also found in Eclipse marketplace that the TM Terminal has this issue on the MacOS. 

See screenshot below :

Some users has also reported that they have fix the issue after update. 
 
![image](https://user-images.githubusercontent.com/1253716/178000137-712dbfcf-b3aa-4a06-97c5-2e11c51cf356.png)


Link to comment : [Click here ](https://marketplace.eclipse.org/comment/6300#comment-6300)

**Comment by i0ntempest** [comment]
I'm using TM Terminal 10.7, v4.x is long outdated. Also python3 executable works just fine:
<img width="630" alt="Screen Shot 2022-07-08 at 13 47 20" src="https://user-images.githubusercontent.com/16017904/178044293-b1d2cf1e-4b7b-441d-833a-4d74ffe5e91c.png">

**Comment by i0ntempest** [comment]
I just noticed it does that even if I open a new local terminal and manually execute jshell command. So yea probably a problem in Eclipse terminal.

EDIT: just noticed something really weird: if I resize Eclipse window (or just the terminal window) with a terminal open, then execute jshell, then it works as expected. Shell variables changed by this resizing are $LINES and $COLUMNS, but exporting them with big values before executing jshell does not fix it. Could be something else hidden deeper in the environment.

**Comment by nileshkhaire** [comment]
Is starting QuickShell and then resizing Eclipse window or terminal fixing the issue ? 

unfortunately I don't have MacOS so I can't try different things. 

Is it possible for you to try it on another Mac ? 

In past one user has tested with MacOS and it was working fine for him.  

And if this is a terminal issue then a ticket can be created in Terminal component.  [Click Here](https://bugs.eclipse.org/bugs/buglist.cgi?component=terminal&product=CDT&resolution=---)

**Comment by i0ntempest** [comment]
Yes, after resizing hit enter once and the prompt will change from `>...` to `jshell>`. I'll open a ticket there, thanks.
